kinto_attachment/utils.py

```diff
@@ -217,3 +217,16 @@ def save_file(content, request, randomize=True, gzipped=False):
     })
 
     return attachment
+
+
+def setting_value(request, name, default):
+    value = request.registry.settings.get('attachment.{}'.format(name), default)
+    if 'bucket_id' in request.matchdict:
+        bid = '/buckets/{bucket_id}'.format(**request.matchdict)
+        if bid in request.registry.attachment_resources:
+            value = request.registry.attachment_resources[bid][name]
```

Comment:
You might want to check also that `name` is in `request.registry.attachment_resources[bid]`.